source/runtime.cpp

```diff
@@ -5015,17 +5036,33 @@ bool reshade::runtime::get_texture_data(api::resource resource, api::resource_us
 				break;
 			case api::format::r10g10b10a2_unorm:
 			case api::format::b10g10r10a2_unorm:
-				for (size_t x = 0; x < pixels_row_pitch; x += 4)
+				// SDR: Quantize the image down to 8-bpc for compatibility with standard screenshot formats
+				if (_back_buffer_color_space != api::color_space::hdr10_st2084)
 				{
-					const uint32_t rgba = *reinterpret_cast<const uint32_t *>(mapped_pixels + x);
-					// Divide by 4 to get 10-bit range (0-1023) into 8-bit range (0-255)
-					pixels[x + 0] = (( rgba & 0x000003FF)        /  4) & 0xFF;
-					pixels[x + 1] = (((rgba & 0x000FFC00) >> 10) /  4) & 0xFF;
-					pixels[x + 2] = (((rgba & 0x3FF00000) >> 20) /  4) & 0xFF;
-					pixels[x + 3] = (((rgba & 0xC0000000) >> 30) * 85) & 0xFF;
-					if (view_format == api::format::b10g10r10a2_unorm)
-						std::swap(pixels[x + 0], pixels[x + 2]);
+					for (size_t x = 0; x < pixels_row_pitch; x += 4)
+					{
+						const uint32_t rgba = *reinterpret_cast<const uint32_t *>(mapped_pixels + x);
+						// Divide by 4 to get 10-bit range (0-1023) into 8-bit range (0-255)
+						pixels[x + 0] = (( rgba & 0x000003FF)        /  4) & 0xFF;
+						pixels[x + 1] = (((rgba & 0x000FFC00) >> 10) /  4) & 0xFF;
+						pixels[x + 2] = (((rgba & 0x3FF00000) >> 20) /  4) & 0xFF;
+						pixels[x + 3] = (((rgba & 0xC0000000) >> 30) * 85) & 0xFF;
+						if (view_format == api::format::b10g10r10a2_unorm)
+							std::swap(pixels[x + 0], pixels[x + 2]);
+					}
 				}
+				// HDR10: Keep the original data, do not convert to 8-bpc
+				else
+					std::memcpy(pixels, mapped_pixels, pixels_row_pitch);
+				break;
+			case api::format::r16g16b16a16_float:
+				// FP16 is implicitly always scRGB
+				if (_back_buffer_format == api::format::r16g16b16a16_float &&
+				    _back_buffer_color_space != api::color_space::extended_srgb_linear)
+				{
+					_back_buffer_color_space = api::color_space::extended_srgb_linear;
```

Comment:
Yes, that can be removed. I was worried that having no colorspace assigned would cause issues with other things, as some software never calls `SetColorSpace1 (...)` for scRGB-based HDR. But it looks like everything handles the implicit colorspace correctly.